consensus/src/aggregation/engine.rs

```diff
@@ -847,14 +843,17 @@ impl<
                     self.pending
                         .insert(index, Pending::Verified(digest, epoch_map));
 
-                    // If the digest is verified but not yet confirmed, set a rebroadcast deadline
-                    self.set_rebroadcast_deadline(index);
+                    // If we've already generated an ack and it isn't yet confirmed, mark for immediate rebroadcast
+                    self.rebroadcast_deadlines
```

Comment:
In practice, this dramatically improved recovery performance during cluster shutdown.